NineTests/NineTests.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2005, 2007-2008 Henri Verbeet
+ * Copyright (C) 2007-2013 Stefan DÃ¶singer(for CodeWeavers)
+ * Copyright (C) 2008 Jason Green(for TransGaming)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* These tests are directly adapted from wine d3d9 tests, thanks to the original authors.
+ * The code was modified in 2016 */
+
+#include <math.h>
+#include <d3d9.h>
+#include "Xnine.h"
+
+#include <unistd.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <math.h>
+
+#include <gtest/gtest.h>
+
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof(*(a)))
+
+#define UINT_PTR int
+
+#define broken(x) 0
+#define ok(c, ...) {{EXPECT_TRUE(c);}}
+#define ok_(file, line) ok
+#define skip(...) {fprintf(stderr, "skip(%u): ", __LINE__); fprintf(stderr, __VA_ARGS__);}
+#define win_skip(...) {fprintf(stderr, "skip(%u): ", __LINE__); fprintf(stderr, __VA_ARGS__);}
+#define trace(...) {fprintf(stderr, "trace(%u): ", __LINE__);fprintf(stderr, __VA_ARGS__);}
+
+#define ZeroMemory(p, n) memset(p, 0, n)
+
+#define todo_nine(a)
+#define todo_wine
+#define todo_wine_if(c) if (c)
+#define SIZE_T size_t
+#define heap_alloc malloc
+#define HeapAlloc(a, b, c) malloc(c)
+#define HeapFree(a, b, c) free(c)
+#define heap_free free
+#define LPDWORD (DWORD*)
```

Comment:
Would you be OK with this change then in winehq?
https://gitlab.winehq.org/wine/wine/-/merge_requests/5324/diffs?commit_id=37a74860e7e3635063d7100e688303d8ef935a68

Reply:
I could drop this here then (and/or the `sed -i "s/LPDWORD/DWORD*/g"` part)

Reply:
In my opinion, you should keep upstream changes to the absolute minimum. If you can work around things on your side (for example by changing your ok/LPDWORD defines), you should.

But it's not my opinion you should worry about. I'm not a Wine maintainer, I just hang around in their channel.

Reply:
IMHO, as I see the https://gitlab.winehq.org/wine/wine/-/commit/39991e759a45760269ece936408cbd241a1a7c53 it probably make (for some unknown reason) to avoid it :laughing: